lib/python/mod_python/cgihandler.py

```diff
@@ -88,22 +88,22 @@ def handler(req):
 
             try:
                 # we do not search the pythonpath (security reasons)
-                fd, path, desc = imp.find_module(module_name, [dir])
-            except ImportError:
+                spec = importlib.util.spec_from_file_location(module_name, os.path.join(dir, file))
+            except (ModuleNotFoundError, ValueError):
                 raise apache.SERVER_RETURN(apache.HTTP_NOT_FOUND)
 
-            # this executes the module
-            imp.load_module(module_name, fd, path, desc)
+            if spec is None:
+                raise apache.SERVER_RETURN(apache.HTTP_NOT_FOUND)
+
+            module = importlib.util.module_from_spec(spec)
+            sys.modules[module_name] = module
+            spec.loader.exec_module(module)
```

Comment:
This logic was based on https://docs.python.org/3/library/importlib.html#importing-a-source-file-directly, right?

Reply:
Yes